language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaDocumentService.java

```diff
@@ -427,8 +429,9 @@ public CompletableFuture<List<PublishDiagnosticsParams>> diagnostics(BallerinaPr
                         LSContextOperation.DOC_DIAGNOSTICS,
                         this.serverContext);
                 DiagnosticsHelper diagnosticsHelper = DiagnosticsHelper.getInstance(this.serverContext);
+                String decodedFileUri = URLDecoder.decode(fileUri, StandardCharsets.UTF_8);
```

Comment:
Shall we declare the `fileUri` variable by decoding it to `utf-8` and remove this `decodedFileUri` variable?

Reply:
Changed